src/styles/shared/components/buttons/mixins.scss

```diff
@@ -2,174 +2,141 @@
- *        [$name] - Shorthand name of colormap to retrieved. Defaults to base
- *          $colors.
- * @return {sass:map}
- */
-@function colormap($name: 'colors') {
-  $result: $colors;
-  @if $name == 'icon' {
-    $result: $IconButton-colors;
-  }
-  @if $name == 'icon--primary' {
-    $result: $IconButton-colors--primary;
-  }
-  @if $name == 'labeled' {
-    $result: $LabeledButton-colors;
-  }
-  @if $name == 'labeled--primary' {
-    $result: $LabeledButton-colors--primary;
-  }
-  @if $name == 'link' {
-    $result: $LinkButton-colors;
-  }
-  @return $result;
+  @content;
 }
 
-// Base mixin for <button> elements. Start here for new <button> classes
-@mixin Button($options: ()) {
+// Base mixin for a button that has text/content and, optionally, an icon.
+// Supports variants and sizes.
+@mixin LabeledButton($options: ()) {
   $defaultOptions: (
-    // What colors should be used for this button's styling?
-    'colormap': $colors,
-    // Should styling be added for "active" and "hover" states for this <button>?
-    'withStates': true,
-    // Should styling be added to support the layout of multiple sub-elements
-    // of this <button>? Not needed if <button> contains only one child.
-    'withLayout': false,
-    // Internal margin around SVG icon
-    'margin': 0.5em,
-    'border-radius': 2px
+    'colormap': c.$LabeledButton-colors,
+    'colormap--light': c.$LabeledButton-colors--light,
+    'colormap--primary': c.$LabeledButton-colors--primary,
+    'colormap--dark': c.$LabeledButton-colors--dark,
+    'withLayout': true,
   );
   $-options: map.merge($defaultOptions, $options);
-  @include _button($options: $-options);
+
+  @include Button($-options);
+
   @content;
 }
 
-// Mixin for --modifier variants on buttons, e.g. `.MyButton--primary`
-@mixin Button--variant($options: ()) {
+// Base mixin for a button styled to look like an <a> link. Supports variants
+// but not pressed/active states at present.
+@mixin LinkButton($options: ()) {
   $defaultOptions: (
-    // Colors to use when styling this variant
-    'colormap': $colors,
-    // Should this variant have styling for "active" and "hover" states?
-    'withStates': true
+    'colormap': c.$LinkButton-colors,
+    'colormap--light': c.$LinkButton-colors--light,
+    'colormap--primary': c.$LinkButton-colors--primary,
+    'colormap--dark': c.$LinkButton-colors--dark,
   );
   $-options: map.merge($defaultOptions, $options);
-  @include _variant($options: $-options);
+  @include Button($-options) {
+    // Lighter font weight for link-styled buttons
+    font-weight: 400;
+    // Add an underline on hover
+    &:hover:not([disabled]) {
+      text-decoration: underline;
+    }
+
+    &--primary {
+      // Primary variant has bolder text (and also different colors)
+      font-weight: 500;
+    }
+  }
+
+  // Remove horizontal padding to allow button to appear flush-left or -right
+  padding: 0;
```

Comment:
This removes vertical padding too, is that intended or should the comment change?